source/guides/postgres-migration.rst

```diff
@@ -0,0 +1,199 @@
+Audits           Action           text -> varchar(512)
+Audits           ExtraInfo        text -> varchar(1024)
+ClusterDiscovery HostName         text -> varchar(512)
+Commands         IconURL          text -> varchar(1024)
+Commands         AutoCompleteDesc text -> varchar(1024)
+Commands         AutoCompleteHint text -> varchar(1024)
+Compliances      Keywords         text -> varchar(512)
+Compliances      Emails           text -> varchar(1024)
+FileInfo         Path             text -> varchar(512)
+FileInfo         ThumbnailPath    text -> varchar(512)
+FileInfo         PreviewPath      text -> varchar(512)
+FileInfo         Name             text -> varchar(256)
+FileInfo         MimeType         text -> varchar(256)
+LinkMetadata     URL              text -> varchar(2048)
+RemoteClusters   SiteURL          text -> varchar(512)
+RemoteClusters   Topics           text -> varchar(512)
+Sessions         DeviceId         text -> varchar(512)
+Systems          Value            text -> varchar(1024)
+UploadSessions   FileName         text -> varchar(256)
+UploadSessions   Path             text -> varchar(512)
+================ ================ =====================
+
+As you can see, there are several occurrences where the schema can differ and data size constraints within the PostgreSQL schema can result in errors. Several reports have been received from our community that ``LinkMetadata`` and ``FileInfo`` tables had some overflows, so we recommend checking these tables in particular. Please do check if your data in the MySQL schema exceeds these limitations. You can check if there are any required deletions. For example, to do so in the ``Audits`` table/``Action`` column; run:
+
+.. code:: sql
+
+   DELETE FROM mattermost.Audits where LENGTH(Action) > 512;
+
+Full-text indexes
+~~~~~~~~~~~~~~~~~
+
+It's possible that some words in the ``Posts`` and ``FileInfo`` tables can exceed the `limits of the maximum token length <https://www.postgresql.org/docs/11/textsearch-limitations.html>`__ for full text search indexing. In these cases, we recommend dropping the ``idx_posts_message_txt`` and ``idx_fileinfo_content_txt`` indexes from the PostgreSQL schema, and creating these indexes after the migration by running following queries:
+
+To drop indexes, run the following commands before the migration:
+
+.. code:: sql
+
+   DROP INDEX IF EXISTS idx_posts_message_txt;
+   DROP INDEX IF EXISTS idx_fileinfo_content_txt;
+
+To re-create indexes, run the following once the migration is completed:
+
+.. code:: sql
+
+   CREATE INDEX IF NOT EXISTS idx_posts_message_txt ON posts USING gin(to_tsvector('english', message));
+   CREATE INDEX IF NOT EXISTS idx_fileinfo_content_txt ON fileinfo USING gin(to_tsvector('english', content));
+
+Migrate the data
+----------------
+
+Once we set the schema to desired state, we can start migrating the **data** by running ``pgLoader`` \*\*
+
+\*\* Use the following configuration for the baseline of the data migration:
```

Comment:
@isacikgoz what are you thoughts on us providing the migration.load template for users to download/edit? Link here maybe?

Reply:
The template is actually on the repo, being used by the workflow: https://github.com/mattermost/mattermost/blob/master/server/tests/template.load we can add the link here.